### PR TITLE
Calculate the SimpleHash for public assets

### PR DESCRIPTION
**Pull request by eccles**
USe api_query instead of explicit params
    
Problem:
Calculating simplehash for public assets is not possible with explicit
 time periods and fqdn.
    
 Solution:
 Simplify the API surface to accept the api_query returned from the
 blockchain endpoint.
    
 Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

**Review by JAG-UK — APPROVED** [review]

File: rkvst_simplehash/v1.py [hunk 1]
```diff
@@ -86,18 +89,15 @@ def redact_event(event):
...
     If page size is specified return the list of records in batches of page_size
     until next_page_token in response is null.
     If page size is unspecified return up to the internal limit of records.
     (different for each endpoint)
     Args:
-        start_time (string): rfc3339 formatted datetime string of the start date of the time window
-                             of events
-        end_time (string): rfc3339 formatted datetime string of the end date of the time window of
-                           events
+        api_query (string): The api_query as returned from the simplehashed event response.
```
Comment (honourfish) [inline rkvst_simplehash/v1.py:100]: minor: simplehash event response

File: rkvst_simplehash/v1.py [hunk 2]
```diff
@@ -246,36 +221,31 @@ def main():  # pragma: no cover
...
+            auth_token = str(fd.read()).strip("\n")
+
+    elif args.client_id and args.client_secret_file:
+        # we don't have the auth token file, but we have a client id and secret
+        # so attempt to get the auth token via client id and secret
+        with open(args.client_secret_file, encoding="utf-8") as fd:
+            client_secret = str(fd.read()).strip("\n")
+            auth_token = get_auth_token(args.api_query, args.client_id, client_secret)
+
+    # auth_token may be None here and this will only work with the publicassets endpoint
```
Comment (honourfish) [inline rkvst_simplehash/v1.py:246]: lovely comment :)

File: scripts/functest.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    rkvst_simplehashv1 \
+        --client-id "${RKVST_APPREG_CLIENT}" \
+        --client-secret-file "${RKVST_APPREG_SECRET_FILENAME}" \
+        "${API_QUERY}"
+else
+    echo "Cannot use appregistration - please set RKVST_APPREG_CLIENT"
+    echo "Cannot use appregistration - please set RKVST_APPREG_SECRET_FILENAME"
+fi
+
+API_QUERY="https://app.dev-paul-0.wild.jitsuin.io"
```
Comment (honourfish) [inline scripts/functest.sh:46]: I suggest we don't use your fqdn for the tests?

File: unittests/constants.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+        "principal_declared": {
+            "issuer": "https://rkvt.com",
+            "subject": "117303158125148247777",
+            "display_name": "William Defoe",
+            "email": "WilliamDefoe@rkvst.com",
+        },
+        "principal_accepted": {
+            "issuer": "https://rkvt.com",
+            "subject": "117303158125148247777",
+            "display_name": "William Defoe",
```
Comment (honourfish) [inline unittests/constants.py:55]: glad to see william staying strong in the tests :P

File: scripts/functest.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+API_QUERY+="/archivist/v2/publicassets/-/events"
+API_QUERY+="?proof_mechanism=SIMPLE_HASH"
+API_QUERY+="&timestamp_accepted_since=2022-10-07T07:01:34Z"
+API_QUERY+="&timestamp_accepted_before=2022-10-16T13:14:56Z"
+API_QUERY+="&order_by=SIMPLEHASHV1"
+
+echo
+echo "Get public assets"
+rkvst_simplehashv1 "${API_QUERY}"
```
Comment (honourfish) [inline scripts/functest.sh:55]: can we have another test that is authenticated (either by the tenant that creates the events or not, and ensure the hash for public assets is correct?

**Pull request merged**